core/io/marshalls.h

```diff
@@ -50,18 +50,18 @@ typedef uint32_t uintr_t;
 
 union MarshallFloat {
 	uint32_t i; ///< int
-	float f; ///< float
+	float f = 0.0f; ///< float
 };
 
 union MarshallDouble {
 	uint64_t l; ///< long long
-	double d; ///< double
+	double d = 0.0f; ///< double
```

Comment:
I'm not sure that it's correct to initialize a union like this. The union may be used with the other type.

In the cases of marshalls specifically, the entire purpose of these unions is to convert the raw bits in the float to an integer for bitwise operations. Therefore, they will always be initialized with a custom value when they are used. The default value will never be read ever.